### PR TITLE
Created tests for SetNamedPipeHandleState

### PR DESCRIPTION
**Pull request by exvito**
While reviewing tests, I found out that the SetNamedPipeHandleState wasn't being tested:
- Created a few very basic tests to ensure the wrapper is checked, at least.
- More complex tests need to be run against Named Pipes which aren't available for now.

### CODE REVIEW

**Comment by exvito** [comment]
Preliminary check results notes:
- As in #91 the READTHEDOCS=1 check for 2.7 is failing.
- AppVeyor failed oddly in 2.6/32 claiming newer queued builds for this PR and then timing out.

Both a mystery. Could you please educate me a bit on these so that I don't repeat this type of mistakes in the future?

Again, thanks.

**Comment by opalmer** [comment]
> As in #91 the READTHEDOCS=1 check for 2.7 is failing.

This should be fixed when #93 merges, you'll just need to pull in the latest master.

> AppVeyor failed oddly in 2.6/32 claiming newer queued builds for this PR and then timing out.

See the updates to appveyor.yml in #83.  Basically instead of building every commit for every PR, appveyor will only build the latest commit now.  The turnaround time for one commit on AppVeyor ranges from thirty minutes to over an hour so this saves a significant amount of waiting if there are multiple commits.

As for the code, tests look good thanks for the PR.  I've updated the name of your PR a bit though to remove the 'Tests -' bit because it will keep the release page a bit more consistent.

**Comment by exvito** [comment]
> See the updates to appveyor.yml in #83. Basically instead of building every commit for every PR, appveyor will only build the latest commit now. The turnaround time for one commit on AppVeyor ranges from thirty minutes to over an hour so this saves a significant amount of waiting if there are multiple commits.

I was aware of both: that AppVeyor is sloooooooow and that at some point you updated the tools to avoid wasting time on commits other than the latest, which is great.

What I found strange was that, at that time, AppVeyor was indeed working with the latest commit. Here is a simplified timestamp analysis (couldn't find timezone info for AppVeyour, must be GMT+1 for this to make any sense):
- 12:20:51 GMT+1 - 1st commit - eb09845.
- 12:31:07 - AppVeyor build 905 - Started 2.6 / 32 - Aborted - Why?!
- 12:36:47 - AppVeyor build 905 - Started 2.6 / 64 - Ok
- 12:47:23 - AppVeyor build 905 - Started 2.7 / 32 - Ok
- 12:51:34 - AppVeyor build 905 - Started 2.7 / 64 - Ok
- 12:57:43 - AppVeyor build 905 - Started 3.3 / 32 - Ok
- 13:00:28 - AppVeyor build 905 - Started 3.3 / 64 - Ok
- 13:01:49 GMT+1 - 2nd commit - dadc608.
- 13:06:14 - AppVeyor build 905 - Started 3.4 / 32 - Aborted
- 13:09:49 - AppVeyor build 905 - Started 3.4 / 64 - Aborted
- 13:13:09 - AppVeyor build 905 - Started 3.5 / 32 - Aborted
- 13:16:50 - AppVeyor build 905 - Started 3.5 / 64 - Aborted

In other words, AppVeyor job 905 aborts after 2nd commit make sense; I just can't find an explanation for the first abort, all I notice is that the log is somewhat different, ending in some kind of "timeout" error.

Without wanting to chase non-critical details, I found that really odd. Can you figure out alternative explanations for the abort on the first job?

Other than that, thanks for #93 and for updating the PR name. I'll be updating next.

**Comment by opalmer** [comment]
> Without wanting to chase non-critical details, I found that really odd. Can you figure out alternative explanations for the abort on the first job?

I'm not entirely sure honestly.  If I had to guess it probably has do to with the data being returned from the underlying APIs.  Overall it seems to work I guess though.

Other than that, change looks good.  Merging.
